src/main/java/org/mastodon/mamut/util/LineageTreeUtils.java

```diff
@@ -67,4 +72,68 @@ public void crossComponent( BranchSpot from, BranchSpot to, DepthFirstSearch< Br
 			search.start( root );
 		}
 	}
+
+	/**
+	 * Performs a depth-first iteration through the specified {@link ModelGraph} starting from the given spot.
+	 * <p>
+	 * The iteration is performed in a single thread. The given {@code action} is called for each vertex (i.e. {@link Spot})
+	 * in the graph, when all its descendants in the search tree have been iterated through or when it is a leaf in the tree.
+	 *
+	 * @param modelGraph the graph to iterate through.
+	 * @param spot       the spot to start the iteration from.
+	 * @param action      the action to perform on each spot.
+	 */
+	public static void callDepthFirst( ModelGraph modelGraph, Spot spot, Consumer< Spot > action )
```

Comment:
You don't need to have two methods for the ModelGraph and the ModelBranchGraph. It can be one method that uses generics.

Reply:
As discussed together. This method is indeed not needed an replaced by using an DepthFirstIterator in ClusterRootNodesController directly. Cf.: https://github.com/mastodon-sc/mastodon-deep-lineage/commit/b67f9ff446c6284c12caae6844377860298b31b3